packages/timezone_map/lib/src/map.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:io';
+
+import 'package:file/local.dart';
+import 'package:file/file.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:intl/intl.dart';
+import 'package:latlong2/latlong.dart';
+import 'package:path/path.dart' as p;
+
+import 'latlng.dart';
+
+/// A widget that displays a map of the world.
+class TimezoneMap extends StatelessWidget {
+  /// Creates a map.
+  const TimezoneMap({
+    super.key,
+    this.marker,
+    this.offset,
+    this.onPressed,
+  });
+
+  /// Coordinates of a map marker.
+  final LatLng? marker;
+
+  /// UTC-offset of the highlighted timezone.
+  final double? offset;
+
+  /// Called when the map is pressed at [coordinates].
+  final void Function(LatLng coordinates)? onPressed;
+
+  /// Requests all timezone map SVG assets to be pre-cached.
+  ///
+  /// **Note**: not supported on Web.
+  static Future<void> precacheAssets(BuildContext context,
+      {@visibleForTesting FileSystem fs = const LocalFileSystem()}) {
+    if (kIsWeb) return Future.value();
+    final exePath = Platform.resolvedExecutable;
+    final assetDir = fs.directory(p.join(p.dirname(exePath), 'data',
+        'flutter_assets', 'packages', 'timezone_map', 'assets'));
+    final assets = <String>[];
+    if (assetDir.existsSync()) {
+      assets.addAll(assetDir
+          .listSync()
+          .map((entity) => p.basename(entity.path))
+          .where((asset) => p.extension(asset) == '.svg'));
+    }
```

Comment:
This method of listing all the svg assets within the package looks rather complicated. I've tested the following locally (only on the example app, not within the context of a third party app like the installer) and it seems to do the job with much less code:

```dart
DefaultAssetBundle.of(context)
    .loadString('AssetManifest.json')
    .then((value) => json.decode(value).keys.where((asset) =>
        p.isWithin('packages/timezone_map', asset) &&
        p.extension(asset) == '.svg'))
    .then(…)
```

Reply:
much nicer, thanks! :+1: i'll give it a try and update the tests